httpclient/httpclient.go

```diff
@@ -63,6 +64,8 @@ type Config struct {
 	Tracer tracer
 	// DialContext allows a dial context to be injected into the HTTP transport.
 	DialContext func(ctx context.Context, network string, addr string) (net.Conn, error)
+	// HTTPTracer allows use of the httptrace library for indepth HTTP logging and debugging
```

Comment:
see the tracer interface above ^^

is this good enough..

https://github.com/circleci/ex/blob/main/httpclient/metrics/metrics.go